REGRESSION(262173@main): media/media-source/media-source-fastseek.html is a flaky text failure

jyavenard opened the pull request:
#### ab74fe9cba783e46bf6dc8d39e98215d383805c3
<pre>
Need a short description (OOPS!).
Need the bug URL (OOPS!).
Include a Radar link (OOPS!).

Reviewed by NOBODY (OOPS!).

Explanation of why this fixes the bug (OOPS!).

* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/Modules/mediasource/MediaSource.cpp:
(WebCore::MediaSource::completeSeek):
</pre>

eric-carlson commented on Source/WebCore/Modules/mediasource/MediaSource.cpp:
```diff
@@ -177,52 +176,47 @@ MediaTime MediaSource::duration() const
 MediaTime MediaSource::currentTime() const
 {
-    return m_mediaElement ? m_mediaElement->currentMediaTime() : MediaTime::zeroTime();
+    return m_pendingSeekTarget ? m_pendingSeekTarget->time : m_mediaElement ? m_mediaElement->currentMediaTime() : MediaTime::zeroTime();
```
Nit: I think nested ternary operators are hard to read, especially when they get so wide you that you have to scroll horizontally to see the whole thing.

jyavenard replied:
broke it a bit

eric-carlson commented on Source/WebCore/Modules/mediasource/MediaSource.cpp:
```diff
@@ -234,24 +228,52 @@ void MediaSource::completeSeek()
+    auto seekTarget = *m_pendingSeekTarget;
+    m_pendingSeekTarget.reset();
+
+    struct CallbackAggregator final : public RefCounted<CallbackAggregator> {
```
We already have <wtf/CallbackAggregator.h> so this should probably have a more specific name.

jyavenard replied:
I've followed the existing pattern, there are multiple CallbackAggregator object in the code, they all replicate the WTF one with some changes.
There's one in WebSiteDataStore, 3 in the NetworkProcess code, and on in CDMInstanceFairPlayStreamingAVFObjC, and a few more with a slight change in the name

jyavenard replied:
I renamed it to `SeeksCallbackAggregator`

eric-carlson commented on Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.h:
```diff
@@ -85,8 +85,8 @@ ALLOW_NEW_API_WITHOUT_GUARDS_END
-    MediaPlayer::NetworkState networkState() const override;
-    MediaPlayer::ReadyState readyState() const override;
+    MediaPlayer::NetworkState networkState() const final;
+    MediaPlayer::ReadyState readyState() const final;
```
These unrelated `override` to `final` changes are a mistake:
- they make it harder to pick out and review the changes necessary for this fix
- they will make future `blame` analysis more complicated
- I believe it is contrary to standard WebKit practice

jyavenard replied:
done

eric-carlson commented on Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaStreamAVFObjC.h:
```diff
@@ -72,16 +72,16 @@ class MediaPlayerPrivateMediaStreamAVFObjC final
+    MediaPlayer::NetworkState networkState() const final;
     void setNetworkState(MediaPlayer::NetworkState);
-    MediaPlayer::ReadyState readyState() const override;
+    MediaPlayer::ReadyState readyState() const final;
```
Ditto.

eric-carlson commented on Source/WebCore/platform/mock/mediasource/MockMediaPlayerMediaSource.h:
```diff
@@ -50,47 +50,45 @@ class MockMediaPlayerMediaSource : public MediaPlayerPrivateInterface, public Ca
+    void pause() final;
+    FloatSize naturalSize() const final;
+    bool hasVideo() const final;
+    bool hasAudio() const final;
```
Ditto

eric-carlson commented on Source/WebCore/platform/mock/mediasource/MockMediaSourcePrivate.h:
```diff
@@ -79,16 +78,14 @@ class MockMediaSourcePrivate final
+    bool isEnded() const final { return m_isEnded; }
+    void unmarkEndOfStream() final;
+    MediaPlayer::ReadyState readyState() const final;
+    void setReadyState(MediaPlayer::ReadyState) final;
```
Ditto

squelart commented on Source/WebCore/platform/graphics/MediaPlayer.cpp:
```diff
@@ -2013,6 +2013,17 @@ String MediaPlayer::lastErrorMessage() const
+{
+    StringBuilder builder;
+    builder.append("[");
+    builder.append(WTF::LogArgument<MediaTime>::toString(time));
+    builder.append(WTF::LogArgument<MediaTime>::toString(negativeThreshold));
+    builder.append(WTF::LogArgument<MediaTime>::toString(positiveThreshold));
```
Shouldn't you also append some separators between these values?

jyavenard replied:
it shows like this `[{"value":2701.603090079002}{"value":2}{"value":5}]` which I thought was good enough as MediaTime already adds `{ }`

squelart replied:
Alright then, good enough for logging/debugging by people in the know, I suppose. 👍

youennf commented on Source/WebCore/Modules/mediasource/MediaSource.cpp:
```diff
@@ -185,73 +186,98 @@ const PlatformTimeRanges& MediaSource::buffered() const
+    m_pendingSeekTarget.reset();
+
+    struct SeeksCallbackAggregator final : public RefCounted<SeeksCallbackAggregator> {
+        explicit SeeksCallbackAggregator(MediaTime target, MediaSource& source, CompletionHandler<void(const MediaTime&)>&& completionHandler)
```
No need for explicit

youennf commented on Source/WebCore/Modules/mediasource/MediaSource.cpp:
```diff
@@ -185,73 +186,98 @@ const PlatformTimeRanges& MediaSource::buffered() const
+
+        ~SeeksCallbackAggregator()
+        {
+            ASSERT(completionHandler);
```
Does not bring much here, maybe add it in constructor instead if useful?

youennf commented on Source/WebCore/Modules/mediasource/MediaSource.cpp:
```diff
@@ -185,73 +186,98 @@ const PlatformTimeRanges& MediaSource::buffered() const
+void MediaSource::seekToTarget(const SeekTarget& target, CompletionHandler<void(const MediaTime&)>&& completionHandler)
 {
     if (isClosed())
         return;
```
Should we `seekCompletedHandler(MediaTime::invalidTime())` if closed?

jyavenard replied:
This is done when the MediaSource readyState changes to closed.

youennf commented on Source/WebCore/Modules/mediasource/MediaSource.cpp:
```diff
@@ -978,6 +1004,9 @@ void MediaSource::detachFromElement(HTMLMediaElement& element)
     m_mediaElement = nullptr;
+
+    if (m_seekCompletedHandler)
+        m_seekCompletedHandler(MediaTime::invalidTime());
```
Is it the only case where we need to call m_seekCompletedHandler?
What about when it gets stopped for instance?

jyavenard replied:
done

youennf commented on Source/WebCore/html/HTMLMediaElement.h:
```diff
@@ -1081,19 +1081,15 @@ class HTMLMediaElement
+        SeekTarget target;
     };
-    std::unique_ptr<PendingSeek> m_pendingSeek;
+    std::optional<PendingSeek> m_pendingSeek;
```
We have done some work to move from optional to unique_ptr to reduce some object's memory footprint.
Do we need this change?

jyavenard replied:
I wasn't aware of this. I would have assumed here that allocation within the HTMMediaElement class is better than a single heap allocation with all the required overhead.

youennf commented on Source/WebCore/platform/graphics/MediaPlayerPrivate.h:
```diff
@@ -20,7 +20,7 @@
  * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
- * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
All these file changes are unrelated.

youennf commented on Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
```diff
@@ -256,6 +256,27 @@ void SourceBufferPrivate::reenqueSamples(const AtomString& trackID)
+void SourceBufferPrivate::seekToTarget(const SeekTarget& target, CompletionHandler<void(const MediaTime&)>&& completionHandler)
+{
+    if (!isAttached())
+        return;
```
Call completionHandler probably.

jyavenard replied:
Ah indeed

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
```diff
@@ -529,81 +521,66 @@ void getSupportedTypes(HashSet<String, ASCIICaseInsensitiveHash>& types) const f
-        m_lastSeekTime = MediaTime::createWithDouble(m_lastSeekTime.toDouble(), MediaTime::DefaultTimeScale);
-
-    m_seekCompleted = WaitingToSeek;
+    auto pendingSeek = WTFMove(*m_pendingSeek);
```
Can we std::exchange?

youennf commented on Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
```diff
@@ -529,81 +521,66 @@ void getSupportedTypes(HashSet<String, ASCIICaseInsensitiveHash>& types) const f
         if (auto player = m_player.get())
             player->timeChanged();
-    }
+    } else
```
Would read better with
```
if () {
    ...
    return;
}
ALWAYS_LOG(...)

youennf commented on Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.cpp:
```diff
@@ -83,12 +83,14 @@ const PlatformTimeRanges& RemoteMediaSourceProxy::buffered() const
+    }
 
-    m_connectionToWebProcess->connection().send(Messages::MediaSourcePrivateRemote::SeekToTime(time), m_identifier);
+    m_connectionToWebProcess->connection().sendWithAsyncReply(Messages::MediaSourcePrivateRemote::SeekToTarget(target), WTFMove(completionHandler), m_identifier);
```
Is MediaTime:: invalidTime() used if GPUProcess crashes in the middle of this async IPC?

jyavenard replied:
I don't know how IPC deals with process crashes with completion handler. This seems to be a more overall problem.

jyavenard replied:
Created a `AsyncReplyError<WTF::MediaTime>` template to define the default value when a connection failure occurs

youennf commented on Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
```diff
@@ -355,13 +355,15 @@ void SourceBufferPrivateRemote::setAppendWindowEnd(const MediaTime& appendWindow
+    }
 
-    gpuProcessConnection->connection().send(Messages::RemoteSourceBufferProxy::SeekToTime(mediaTime), m_remoteSourceBufferIdentifier);
+    gpuProcessConnection->connection().sendWithAsyncReply(Messages::RemoteSourceBufferProxy::SeekToTarget(target), WTFMove(completionHandler), m_remoteSourceBufferIdentifier);
```
Ditto, invalidTime is probably what we want if GPUProcess crashes within the async call.

jyavenard replied:
I've created a `AsyncReplyError<WTF::MediaTime>` that sets the default return value to `MediaTime::invalidTime()`

thanks for the pointers

youennf reviewed (approved):
r=me once bots are happy

jyavenard committed:
REGRESSION(262173@main): media/media-source/media-source-fastseek.html is a flaky text failure
https://bugs.webkit.org/show_bug.cgi?id=260185
rdar://113881009

Reviewed by Youenn Fablet.

fastSeek and normal Seek followed different paths and the flow between
GPU process and content process was disconnected. This leads to a racy
behaviour where the HTMLMediaElement in the content process could assume
that the seek operation in the GPU process had completed when it hadn't.
There's been a few workarounds put in place for this problem but none
fixed the core issue: we can only complete the seek operation once the
media source has received all the data and we must wait rather than
expect the player to keep retrying.

fastSeek time target calculation was also left to each of the MediaPlayerPrivate
to handle.
This change move part of the seek operation back to the content process,
where the MediaSource element will ensure that there is data to continue
the seek.
It will also handle the fastSeek calculation.
Once the MediaSource has seeked each of the SourceBuffer and has determined
the final time destination, a CompletionHandler will be resolved with the
exact to seeked to location. From that point, each MediaPlayerPrivate can
resume the seek operation and once complete notify the HTMLMediaElement so
that it can fire the appropriate events.

This also gets us closer to having the handling of the SourceBuffer's content
back to the content process, so that all unsafe demuxing operations can
be performed there.

In this change, the various seek methods (from float, double, MediaTime),
accurate and fast are combined into a single seekToTarget with a new
SeekTarget object that can be used for all types.
We remove the need for SourceBufferPrivate and MediaSourcePrivate's having
to track if it is currently seeking or not and the need for
waitForSeekCompleted/willSeek/seekCompleted calls.

Platform specific changes:
- MediaPlayerPrivateMediaSourceAVFObjC: the seeking internal logic was broken.
m_synchronizerSeeking which is supposed to monitor the AVSampleBufferRenderSynchronizer
seeking state was always set to false. As such, we would fire the timeupdate/seeked event even
before the proper frame was displayed and the new time was set.
This likely explains several intermittent failures with video reftest.
- MediaPlayerPrivateGStreamerMSE: The structure of the code doesn’t permit an easy migration
to the new asynchronous flow. This is tracked in bug 260607 to be done at a later stage.

Covered by existing tests.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Source/WebCore/Modules/mediasource/MediaSource.cpp:
(WebCore::MediaSource::MediaSource):
(WebCore::MediaSource::currentTime const):
(WebCore::MediaSource::seekToTarget):
(WebCore::MediaSource::completeSeek):
(WebCore::MediaSource::monitorSourceBuffers):
(WebCore::MediaSource::detachFromElement):
(WebCore::MediaSource::seekToTime): Deleted.
* Source/WebCore/Modules/mediasource/MediaSource.h:
(WebCore::MediaSource::isSeeking const):
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::seekToTarget):
(WebCore::SourceBuffer::seekToTime): Deleted.
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::setReadyState):
(WebCore::HTMLMediaElement::fastSeek):
(WebCore::HTMLMediaElement::seek):
(WebCore::HTMLMediaElement::seekInternal):
(WebCore::HTMLMediaElement::seekWithTolerance):
(WebCore::HTMLMediaElement::seekTask):
(WebCore::HTMLMediaElement::setCurrentTimeWithTolerance):
* Source/WebCore/html/HTMLMediaElement.h:
(WebCore::HTMLMediaElement::PendingSeek::PendingSeek):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::MediaPlayer::seekToTarget):
(WebCore::MediaPlayer::seekToTime):
(WebCore::MediaPlayer::seekWhenPossible):
(WebCore::MediaPlayer::readyStateChanged):
(WebCore::SeekTarget::toString const):
(WebCore::MediaPlayer::seekWithTolerance): Deleted.
(WebCore::MediaPlayer::seek): Deleted.
* Source/WebCore/platform/graphics/MediaPlayer.h:
(WebCore::SeekTarget::SeekTarget):
(WebCore::SeekTarget::zero):
(WTF::LogArgument<WebCore::SeekTarget>::toString):
* Source/WebCore/platform/graphics/MediaPlayerPrivate.h:
(WebCore::MediaPlayerPrivateInterface::hasClosedCaptions const):
(WebCore::MediaPlayerPrivateInterface::seek): Deleted.
(WebCore::MediaPlayerPrivateInterface::seekDouble): Deleted.
(WebCore::MediaPlayerPrivateInterface::seekWithTolerance): Deleted.
* Source/WebCore/platform/graphics/MediaSourcePrivate.h:
(WebCore::MediaSourcePrivate::setIsSeeking): Deleted.
(WebCore::MediaSourcePrivate::isSeeking const): Deleted.
* Source/WebCore/platform/graphics/MediaSourcePrivateClient.h:
* Source/WebCore/platform/graphics/SourceBufferPrivate.cpp:
(WebCore::SourceBufferPrivate::seekToTarget):
(WebCore::SourceBufferPrivate::fastSeekTimeForMediaTime): Deleted.
* Source/WebCore/platform/graphics/SourceBufferPrivate.h:
(WebCore::SourceBufferPrivate::setMaximumQueueDepthForTrackID):
* Source/WebCore/platform/graphics/avfoundation/MediaPlayerPrivateAVFoundation.cpp:
(WebCore::MediaPlayerPrivateAVFoundation::seekToTarget):
(WebCore::MediaPlayerPrivateAVFoundation::seek): Deleted.
(WebCore::MediaPlayerPrivateAVFoundation::seekWithTolerance): Deleted.
* Source/WebCore/platform/graphics/avfoundation/MediaPlayerPrivateAVFoundation.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::seekToTargetInternal):
(WebCore::MediaPlayerPrivateAVFoundationObjC::seekToTime): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.h:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::PendingSeek::PendingSeek): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::convertEnumerationToString):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::MediaPlayerPrivateMediaSourceAVFObjC):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::seekToTarget):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::seekInternal):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::seekCompleted):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::seeking const):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::setHasAvailableVideoFrame):
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::seekWithTolerance): Deleted.
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::waitForSeekCompleted): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaStreamAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaSourcePrivateAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaSourcePrivateAVFObjC.mm:
(WebCore::MediaSourcePrivateAVFObjC::seekToTarget):
(WebCore::MediaSourcePrivateAVFObjC::waitForSeekCompleted): Deleted.
(WebCore::MediaSourcePrivateAVFObjC::seekCompleted): Deleted.
(WebCore::MediaSourcePrivateAVFObjC::seekToTime): Deleted.
(WebCore::MediaSourcePrivateAVFObjC::fastSeekTimeForMediaTime): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.h:
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
(WebCore::MediaPlayerPrivateWebM::play):
(WebCore::MediaPlayerPrivateWebM::seekToTarget):
(WebCore::MediaPlayerPrivateWebM::seek): Deleted.
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::MediaPlayerPrivateGStreamer):
(WebCore::MediaPlayerPrivateGStreamer::play):
(WebCore::MediaPlayerPrivateGStreamer::doSeek):
(WebCore::MediaPlayerPrivateGStreamer::seekToTarget):
(WebCore::MediaPlayerPrivateGStreamer::updatePlaybackRate):
(WebCore::MediaPlayerPrivateGStreamer::currentMediaTime const):
(WebCore::MediaPlayerPrivateGStreamer::playbackPosition const):
(WebCore::MediaPlayerPrivateGStreamer::finishSeek):
(WebCore::MediaPlayerPrivateGStreamer::updateStates):
(WebCore::MediaPlayerPrivateGStreamer::seek): Deleted.
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.h:
* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
(WebCore::MediaPlayerPrivateGStreamerMSE::seekToTarget):
(WebCore::MediaPlayerPrivateGStreamerMSE::doSeek):
(WebCore::MediaPlayerPrivateGStreamerMSE::seek): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.h:
* Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.cpp:
(WebCore::MediaSourcePrivateGStreamer::seekToTarget):
(WebCore::MediaSourcePrivateGStreamer::seekCompleted): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.h:
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.cpp:
(WebCore::SourceBufferPrivateGStreamer::isSeeking const): Deleted.
* Source/WebCore/platform/graphics/gstreamer/mse/SourceBufferPrivateGStreamer.h:
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
(WebCore::MediaPlayerPrivateMediaFoundation::seekToTarget):
(WebCore::MediaPlayerPrivateMediaFoundation::seek): Deleted.
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.h:
* Source/WebCore/platform/mock/mediasource/MockMediaPlayerMediaSource.cpp:
(WebCore::MockMediaPlayerMediaSource::seekToTarget):
(WebCore::MockMediaPlayerMediaSource::seekWithTolerance): Deleted.
(WebCore::MockMediaPlayerMediaSource::waitForSeekCompleted): Deleted.
(WebCore::MockMediaPlayerMediaSource::seekCompleted): Deleted.
* Source/WebCore/platform/mock/mediasource/MockMediaPlayerMediaSource.h:
(WebCore::MockMediaPlayerMediaSource::mediaPlayerLogIdentifier): Deleted.
(WebCore::MockMediaPlayerMediaSource::mediaPlayerLogger): Deleted.
* Source/WebCore/platform/mock/mediasource/MockMediaSourcePrivate.cpp:
(WebCore::MockMediaSourcePrivate::seekToTarget):
(WebCore::MockMediaSourcePrivate::waitForSeekCompleted): Deleted.
(WebCore::MockMediaSourcePrivate::seekCompleted): Deleted.
(WebCore::MockMediaSourcePrivate::seekToTime): Deleted.
* Source/WebCore/platform/mock/mediasource/MockMediaSourcePrivate.h:
* Source/WebCore/platform/mock/mediasource/MockSourceBufferPrivate.cpp:
(WebCore::MockSourceBufferPrivate::isSeeking const): Deleted.
* Source/WebCore/platform/mock/mediasource/MockSourceBufferPrivate.h:
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp:
(WebKit::RemoteMediaPlayerProxy::seekToTarget):
(WebKit::RemoteMediaPlayerProxy::updateCachedState):
(WebKit::RemoteMediaPlayerProxy::seek): Deleted.
(WebKit::RemoteMediaPlayerProxy::seekWithTolerance): Deleted.
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.h:
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.messages.in:
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.cpp:
(WebKit::RemoteMediaSourceProxy::seekToTarget):
(WebKit::RemoteMediaSourceProxy::seekToTime): Deleted.
(WebKit::RemoteMediaSourceProxy::setIsSeeking): Deleted.
(WebKit::RemoteMediaSourceProxy::waitForSeekCompleted): Deleted.
(WebKit::RemoteMediaSourceProxy::seekCompleted): Deleted.
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.h:
* Source/WebKit/GPUProcess/media/RemoteMediaSourceProxy.messages.in:
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.cpp:
(WebKit::RemoteSourceBufferProxy::seekToTarget):
(WebKit::RemoteSourceBufferProxy::seekToTime): Deleted.
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.h:
* Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.messages.in:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::seekToTarget):
(WebKit::MediaPlayerPrivateRemote::timeChanged):
(WebKit::MediaPlayerPrivateRemote::seeking const):
(WebKit::MediaPlayerPrivateRemote::seek): Deleted.
(WebKit::MediaPlayerPrivateRemote::seekWithTolerance): Deleted.
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
(WebKit::MediaSourcePrivateRemote::seekToTarget):
(WebKit::MediaSourcePrivateRemote::setIsSeeking): Deleted.
(WebKit::MediaSourcePrivateRemote::waitForSeekCompleted): Deleted.
(WebKit::MediaSourcePrivateRemote::seekCompleted): Deleted.
(WebKit::MediaSourcePrivateRemote::seekToTime): Deleted.
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.h:
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.h:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaPlayerState.serialization.in:
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp:
(WebKit::SourceBufferPrivateRemote::seekToTarget):
(WebKit::SourceBufferPrivateRemote::seekToTime): Deleted.
* Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.h:

Canonical link: https://commits.webkit.org/267279@main
